### DISCUSSION TITLE
Output the dependency status file as JSON

### DISCUSSION THREAD

**Discussion by funnan-dev** [microsoft/vcpkg#17811] [Ideas]
****

**Is your feature request related to a problem? Please describe.**
In my industry there is currently a need to keep track of the software versions of all dependencies for a given project. VCPKG, fortunately, provides such a list as a `status` text file located at `./vcpkg/installed/vcpkg/status`. Unfortunately, this output is not written in an industry standard format. Which means I require writing my own custom parser to parse this dependency list in order to generate documentation for my clients.

Another issue that I am having is the `status` file doesn't exist in `./installed/vcpkg` when I run `./vcpkg export` it only exists in the original directory I clone vcpkg into. I need this dependency file exported with the libraries so that each snapshot of vcpkg I make for a given project can trace its software dependencies.

Finally, part of tracing software dependencies is knowing their original source location. Currently, this can be found in the ports' `CONTROL` file labeled as Homepage. In order for me to successfully parse and provide the correct information to my clients, I need access to this Homepage from within the `status` file as well.

**Proposed solution**
I propose that this file, at a minimum, is generated as `status.json` or `dependencies.json` so that users of vcpkg may parse this file with currently available json parsers.
I propose that this `status.json` file be included in the export list for `./vcpkg export`
I propose that the `CONTROL` file Homepage be included as part of the `status.json` dependencies.

**Describe alternatives you've considered**
My current work around is to write my own custom text parser to parse out the dependencies from the `status` file. Then, I need to take that dependency list and do a name match of the ports so that I may parse the `CONTROL` files for each Homepage. Finally, during the pipeline stage of my vcpkg, copy the `status` file into the `vcpkg-export.zip` file that my pipeline produces.

**Additional context**
The end goal is to parse the vcpkg software dependencies so that I may present the data to my clients as needed.

**Comment by cbezault** [comment]
I really like the sound of this proposal.
